src/System.Security.Cryptography.Csp/tests/CspParametersTests.cs

```diff
@@ -33,5 +34,89 @@ public static void SetFlags_ValidatesInput()
             Assert.Throws<ArgumentException>(
                 () => cspParameters.Flags = (CspProviderFlags)0x0100 | CspProviderFlags.NoPrompt);
         }
+
+        [Fact]
+        public static void KeyPassword_SetGet()
+        {
+            var cspParameters = new CspParameters();
+            using (var pwd = new SecureString())
+            {
+                pwd.AppendChar('p');
+                cspParameters.KeyPassword = pwd;
+                Assert.Same(pwd, cspParameters.KeyPassword);
+            }
+        }
+
+        [Theory(Skip = "Manual test - requires Smart Card - read instructions")]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void KeyPassword_SmartCard_Manual_Test(bool correctPassword)
+        {
+            // Find info about your smart card:
+            // > certutil -scinfo -silent
+            //   You should see something like:
+            //   Provider = Microsoft Base Smart Card Crypto Provider
+            //   Key Container = {123ABCDE-1234-ABCD-1234-ABCD1234ABCD}
+            // Fill consts below
+            // Run this test - enter correct/incorrect password for your smart card when asked
+            const string provider = "Microsoft Base Smart Card Crypto Provider";
+            const string container = "{123ABCDE-1234-ABCD-1234-ABCD1234ABCD}";
+
+            Console.Write($"Enter {(correctPassword ? "correct" : "incorrect")} password: ");
+
+            using (var pwd = ReadPassword())
+            {
+                var cspParameters = new CspParameters(1, provider, container)
+                {
+                    KeyNumber = (int)KeyNumber.Exchange,
+                    Flags = CspProviderFlags.UseExistingKey | CspProviderFlags.UseMachineKeyStore,
+                    KeyPassword = pwd
+                };
+
+                Action sign = () =>
+                {
+                    using (var rsa = new RSACryptoServiceProvider(cspParameters))
+                    {
+                        var signed = rsa.SignData(new byte[3] { 1, 2, 3 }, "sha256");
+                        Console.WriteLine(Convert.ToBase64String(signed));
+                    }
+                };
+
+                if (correctPassword)
+                    sign();
+                else
+                    Assert.ThrowsAny<CryptographicException>(sign);
+            }
+        }
+
+        static SecureString ReadPassword()
```

Comment:
Even if this code is effectively uncalled it should still have explicit access modifiers.